app.go

```diff
@@ -131,6 +131,27 @@ func (io invokeOption) String() string {
 	return fmt.Sprintf("fx.Invoke(%s)", strings.Join(items, ", "))
 }
 
+// Error registers any number of errors with the application to short-circuit
+// startup. If more than one error is given, the errors are combined into a
+// single error, where each of the error messages are delimited by a newline.
+//
+// Similar to invocations, errors are applied in order. All Provide options
+// registered after an Error option will not be applied. Similarly, no
+// invokes will be run, including those that were registered before an Error.
+func Error(errs ...error) Option {
+	return errOption(errs)
+}
```

Comment:
I think this is simple enough that we can just use `optionFunc` instead of creating a new type.

Reply:
Yea, now that we aren't using `String`, that sounds good.